components/admin/data/widgets/table/component.js

```diff
@@ -0,0 +1,246 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// utils
+import debounce from 'lodash/debounce';
+
+import { fetchWidgets } from 'services/widget';
+
+// Selectors
+
+// Components
+import Spinner from 'components/ui/Spinner';
+import CustomTable from 'components/ui/customtable/CustomTable';
+import SearchInput from 'components/ui/SearchInput';
+
+// Table components
+import EditAction from './actions/EditAction';
+import DeleteAction from './actions/DeleteAction';
+
+// TDs
+import TitleTD from './td/TitleTD';
+import PublishedTD from './td/PublishedTD';
+import OwnerTD from './td/OwnerTD';
+
+// constants
+import { INITIAL_PAGINATION } from 'components/datasets/table/constants';
+
+class WidgetsTable extends React.Component {
+  static propTypes = {
+    user: PropTypes.object.isRequired,
+    dataset: PropTypes.string
+  };
+
+  static defaultProps = { dataset: '' };
+
+
+  state = {
+    pagination: INITIAL_PAGINATION,
+    loading: true,
+    widgets: [],
+    filters: { name: null }
+  }
+
+  componentDidMount() {
+    const { pagination } = this.state;
+
+    fetchWidgets({
+      includes: 'widget,layer,metadata,vocabulary,user',
```

Comment:
That's this includes makes sense? I don't think you can't include widgets inside a widget and so on.. 🤔